chain/chain/src/chain.rs

```diff
@@ -1365,8 +1365,9 @@ impl Chain {
         orphan_misses_chunks: &mut dyn FnMut(OrphanMissingChunks),
         on_challenge: &mut dyn FnMut(ChallengeBody),
     ) -> Result<Option<Tip>, Error> {
+        let block_height = block.header().height();
         let _span =
-            tracing::debug_span!(target: "chain", "Process block", "#{}", block.header().height())
+            tracing::debug_span!(target: "chain", "Process single block", height=block_height)
```

Comment:
We should largely prefer to include code to compute the fields inside the macro. These macros then expand to something like

```
static THIS_SPAN_ENABLED: bool = true;
if THIS_SPAN_ENABLED {
    let fields = Fields::new(...);
}
```

and so computation of `block_height`, however cheap gets skipped entirely if this span would get filtered out.

Reply:
Totally agree with you, but in this particular case `block_height` is used further below.